### PR TITLE
Ft/more s3 locations support

### PR DESCRIPTION
**Pull request by alexanderchan-scality**
Adds more support options for S3 location constraints:
+ optional versioning for S3 backend
+ path style

### CODE REVIEW

File: lib/management.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+const { URL } = require('url');
```
Comment (philipyoo) [inline lib/management.js]: I don't think `url.URL` is supported in Node v6. Instead there is a property: `url.Url` but it doesn't seem to work as intended.

Based on this blog: https://www.contentful.com/blog/2016/11/02/node-v7-urls-deprecations-warnings-and-developer-experience/
Looks like support was added in v7?

Comment (alexanderchan-scality) [reply lib/management.js]: https://nodejs.org/docs/latest-v6.x/api/url.html#url_the_whatwg_url_api
it is also added in 6.13

File: lib/management.js [hunk 2]
```diff
@@ -111,11 +115,23 @@ function patchConfiguration(instanceId, newConf, log, cb) {
                         };
                     }
                     break;
-                case 'location-do-spaces-v1':
+                case 'location-scality-ring-s3-v1':
+                    pathStyle = true; // fallthrough
```
Comment (jonathan-gramain) [inline lib/management.js]: I would recommend not relying on fallthrough capability of `switch`, instead using a map of properties per location name (`{ 'location-scality-bleh': { pathStyle: true, ...} }`) would be more readable and less error-prone.

**Comment by alexanderchan-scality** [comment]
@ironman-machine try
RUN_BACKBEAT_CRR_TESTS=true

**Comment by alexanderchan-scality** [comment]
@ironman-machine 
RUN_BACKBEAT_CRR_TESTS=true
try

File: lib/management/configuration.js [hunk 3]
```diff
@@ -115,11 +119,23 @@ function patchConfiguration(instanceId, newConf, log, cb) {
                         };
                     }
                     break;
-                case 'location-do-spaces-v1':
+                case 'location-scality-ring-s3-v1':
+                    pathStyle = true; // fallthrough
```
Comment (jonathan-gramain) [inline lib/management/configuration.js]: Reporting the style comment about using a mapping object rather than relying on fallthrough of a switch.

Comment (RachedBenMustapha) [reply lib/management/configuration.js]: It's a good cleanup, we'll follow up on it

**Pull request merged**